File: Assets/StreamingAssets/Lua/LCallC/LuaCallObject.lua
print("Lua 调用Object")

--通过调用构造函数创建对象
local Npc = CS.Npc("MeiMei")
Npc.Hp = 100;
print(Npc.Name)
print(Npc.Hp)

--表调用表成员变量 :  C#中的this 
--注意： 对象引用成员变量时，会印象调用this,相当于Lua的self
print(Npc:Output())

--Lua实例化GameObject   Unity中会创建出一个LuaCreateGo物体 坐标为10,10,20
local Tag = CS.UnityEngine.GameObject("LuaCreateGo")
Tag.transform.position = CS.UnityEngine.Vector3(10,10,20)
--注意是:  调用成员方法
Tag:AddComponent(typeof(CS.UnityEngine.BoxCollider)); 

File: Assets/My/Scripts/LCallC/LuaCallObject.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;

public class Npc
{
    public string Name;
    public int Hp { get; set; }

    public Npc()
    {

    }

    public Npc(string name)
    {
        Name = name;
    }

    public string Output()
    {
        return Name;
    }
}

public class LuaCallObject : MonoBehaviour
{
    void Start()
    {
        xLuaEnv.Instance.Dostring("require('LCallC/LuaCallObject')");
        this.gameObject.AddComponent(typeof(BoxCollider));
    }

    private void OnDestroy()
    {
        xLuaEnv.Instance.Free();
    }
}
